### PR TITLE
[#170333051] Use io-spid-commons for spid login implementation

### PR DESCRIPTION
**Pull request by BurnedMarshal**

### CODE REVIEW

File: src/controllers/authenticationController.ts [hunk 1]
```diff
@@ -109,10 +103,16 @@ export default class AuthenticationController {
         newProfile
       );
       if (createProfileResponse.kind !== "IResponseSuccessJson") {
-        return createProfileResponse;
+        // TODO: Must be extended IAuthenticationController to support IResponseErrorTooManyRequests?
+        return createProfileResponse.kind === "IResponseErrorTooManyRequests"
+          ? ResponseErrorInternal("To many requests")
```
Comment (gunzip) [inline src/controllers/authenticationController.ts]: Why this changes ?

```suggestion
          ? ResponseErrorInternal("Too many requests")
```

Comment (BurnedMarshal) [reply src/controllers/authenticationController.ts]: Because the actual implementation of `IAuthenticationController` doesn't support  `IResponseErrorTooManyRequests` as accepted response for the `acs` method.

**Comment by digitalcitizenship** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Please include a description of your PR changes.
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  Could not get info from npm on <a href='https://www.npmjs.com/package/@pagopa/io-spid-commons'><code>&#64;pagopa/io-spid-commons</code></a></a>
  </td>
    </tr>
  </tbody>
</table>



## Affected stories

  * ⚙️ [#170333051](https://www.pivotaltracker.com/story/show/170333051): Utilizzare la libreria io-spid-commons per la gestione delle route di login di SPID


New dependencies added: <a href='https://www.npmjs.com/package/@pagopa/io-spid-commons'><code>&#64;pagopa/io-spid-commons</code></a>.
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/controllers/authenticationController.ts [hunk 2]
```diff
@@ -109,10 +103,16 @@ export default class AuthenticationController {
...
       if (createProfileResponse.kind !== "IResponseSuccessJson") {
-        return createProfileResponse;
+        // TODO: Must be extended IAuthenticationController to support IResponseErrorTooManyRequests?
+        return createProfileResponse.kind === "IResponseErrorTooManyRequests"
+          ? ResponseErrorInternal("Too many requests")
+          : createProfileResponse;
       }
     } else if (getProfileResponse.kind !== "IResponseSuccessJson") {
-      return getProfileResponse;
+      // TODO: Must be extended IAuthenticationController to support IResponseErrorTooManyRequests?
```
Comment (gunzip) [inline src/controllers/authenticationController.ts]:
```suggestion
      // TODO: IAuthenticationController type needs to support IResponseErrorTooManyRequests
```

File: src/controllers/authenticationController.ts [hunk 3]
```diff
@@ -109,10 +103,16 @@ export default class AuthenticationController {
         newProfile
       );
       if (createProfileResponse.kind !== "IResponseSuccessJson") {
-        return createProfileResponse;
+        // TODO: Must be extended IAuthenticationController to support IResponseErrorTooManyRequests?
```
Comment (gunzip) [inline src/controllers/authenticationController.ts]:
```suggestion
      // TODO: IAuthenticationController type needs to support IResponseErrorTooManyRequests
```
